File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        n = min(len(word1), len(word2))
        ans = ''
        for i in range(n):
            ans += word1[i] + word2[i]
        if len(word1) > len(word2):
            ans += word1[n:] 
        elif len(word2) > len(word1):
            ans += word2[n:]
        return ans

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string ans = ""; 
        int p1 = 0, p2 = 0;
        int m = word1.length(), n = word2.length();
        while (p1 < m && p2 < n) {
            ans += word1[p1++];
            ans += word2[p2++];
        }
        while (p1 < m) {
            ans += word1[p1++];
        }
        while (p2 < n) {
            ans += word2[p2++];
        }
        return ans;
    }
};